src/ProjectTemplates/Web.ProjectTemplates/content/SignalR-CSharp/Hubs/StreamingTimerHub.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Company.WebApplication1
+{
+    public class StreamingTimerHub : Hub
+    {
+        public ChannelReader<string> ServerTimer(CancellationToken token)
+        {
+            var channel = Channel.CreateUnbounded<string>();
+            _ = WriteDateAsync(channel.Writer, token);
+            return channel.Reader;
+        }
+
+        private async Task WriteDateAsync(ChannelWriter<string> writer, 
+            CancellationToken token)
+        {
+            try
+            {
+                while(true)
+                {
+                    token.ThrowIfCancellationRequested();
```

Comment:
This doesn't look like a good practice - using exceptions for flow control.

How about:

```c#
if (token.IsCancellationRequested)
{
    break;
}
```